app/src/main/java/com/trianglz/weatherapp/di/modules/ApiModule.kt

```diff
@@ -0,0 +1,55 @@
+package com.trianglz.weatherapp.di.modules
+
+import com.trianglz.weatherapp.BuildConfig
+import com.trianglz.weatherapp.data.apiservice.apininja.ApiNinja
+import com.trianglz.weatherapp.data.apiservice.restcountries.RestCountriesAPI
+import dagger.Module
+import dagger.Provides
+import dagger.hilt.InstallIn
+import dagger.hilt.components.SingletonComponent
+import retrofit2.Retrofit
+import retrofit2.converter.gson.GsonConverterFactory
+import javax.inject.Named
+import javax.inject.Singleton
+
+@Module
+@InstallIn(SingletonComponent::class)
+class ApiModule {
```

Comment:
A DI module will be provided for each data type: cities, countries and weather. To achieve separation.